smoke-tests-otel-starter/spring-boot-3/src/test/java/io/opentelemetry/smoketest/AbstractMongodbSpringStarterSmokeTest.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.smoketest;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import io.opentelemetry.api.trace.SpanKind;
+import io.opentelemetry.semconv.HttpAttributes;
+import io.opentelemetry.semconv.UrlAttributes;
+import io.opentelemetry.semconv.incubating.DbIncubatingAttributes;
+import io.opentelemetry.spring.smoketest.AbstractSpringStarterSmokeTest;
+import io.opentelemetry.spring.smoketest.OtelSpringStarterSmokeTestApplication;
+import io.opentelemetry.spring.smoketest.OtelSpringStarterSmokeTestController;
+import io.opentelemetry.spring.smoketest.SpringSmokeOtelConfiguration;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.web.client.TestRestTemplate;
+
+@SpringBootTest(
+    classes = {OtelSpringStarterSmokeTestApplication.class, SpringSmokeOtelConfiguration.class},
+    webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+abstract class AbstractMongodbSpringStarterSmokeTest extends AbstractSpringStarterSmokeTest {
+
+  @Autowired private TestRestTemplate testRestTemplate;
+
+  @Test
+  void mongodb() {
+    testing.clearAllExportedData(); // ignore data from application startup
```

Comment:
Not sure this line is needed. Each test class should have its own memory exporters from what I understand.

Reply:
Yes, each test has it's own memory exporter - even every test method.

The problem (or feature) is that the exporter is initialized before spring starts: https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/5cc3362ed0f845354342bcad6b40bbca0413248f/smoke-tests-otel-starter/spring-smoke-testing/src/main/java/io/opentelemetry/spring/smoketest/AbstractSpringStarterSmokeTest.java#L27 

There's no other callback just before the test.

This is the span that is cleared, btw:

`name=io.opentelemetry.jdbc, version=2.4.0-alpha-SNAPSHOT, schemaUrl=null, attributes={}}, name=CREATE table db, kind=CLIENT`

Reply:
I would suggest 2 options to improve the code maintability:
* Update the comment at line 32 by mentioning `name=io.opentelemetry.jdbc, version=2.4.0-alpha-SNAPSHOT, schemaUrl=null, attributes={}}, name=CREATE table db, kind=CLIENT`
* Don't execute the code related to the SQL datasource with the MongoDB one

Reply:
fixed by creating the table only in jdbc test